File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0;
        int j = size(numbers)-1;

        while (i < j) {
            auto s = numbers[i] + numbers[j];

            if (s > target) --j;
            else if (s < target) ++i;
            else return {i + 1, j + 1};
        }
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        number_to_index = {nums[i]: i for i in range(len(nums))} # O(n)
        
        for i in range(len(nums)): # O(n)
            ni = nums[i]
            npair =  target - ni
            if npair in number_to_index: 
                j = number_to_index[npair]
                if i !=j :
                    return [i, number_to_index[npair]]